app/api/campaigns/index.js

```diff
@@ -22,70 +22,43 @@ router.get('/', async (req, res) => {
   }
 });
 
-router.get('/:id', (req, res) => {
+router.get('/:id', async (req, res) => {
   const { id } = req.params;
-  Campaigns.findCampaign(id)
-    .then((result) => {
-      if (result) {
-        return Campaigns.findById(id);
-      }
-      res.status(400).json({ msg: 'Campaign was not found in the database' });
-    })
-    .then(async (campaign) => {
-      // If this campaign belongs to a deactivated account, then
-      // only an admin should be able to see it
-
-      if (campaign.is_deactivated) {
-        const user = await Users.findBySub(req.user.sub);
-        if (!user.admin) {
-          return res
-            .status(401)
-            .json({
-              msg: 'This campaign may only be viewed by an administrator',
-            });
-        }
+
+  try {
+    let campaign = await Campaigns.findCampaign(id);
+    if (!campaign) return res.status(400).json({ msg: 'Campaign was not found in the database' });
+    campaign = await Campaigns.findById(id);
```

Comment:
`findCampaign`, then `findById`? And both take an ID? am confused

Reply:
`findCampaign` gets the basic campaign object so it's used as a check to see if the campaign exists, and then `findbyId` gets the rest (e.g. updates, comments). A little confusing for sure but I'll see what I can do since the functions are used in different places for different things.